### ISSUE TITLE
Non-replicable campus inference

### ISSUE THREAD

**Issue by iflab-felipe** [chengche6230/ReST#14]
Hi. I trained my TG and used your pretrained SG (for CAMPUS Garden 1) as another issue suggested (#6) and I cannot replicate your results. I am using osnet_x1_0 as my ReID model. IDs are changing constantly for every frame and person, even though the spatial graph seems ok, as if the SG was working but not the TG. In this sense I also have to say that the results when using both of your pretrained weights is worse than this mixed approach somehow. Results obtained along with an iteration example and the SG:

>  2024-04-12 13:02:16.769 | INFO     | src.datasets.dataset:__getInference__:312 - Loaded graph with 39 nodes and 1108 edges.
2024-04-12 13:02:16.802 | INFO     | src.utils.tracklet:postprocessing_sg:133 - Post-processing done, removed 5 edges. Graph: 39 nodes, 30 edges.
2024-04-12 13:02:17.011 | INFO     | src.tracker:_test_one_epoch:233 - Iteration 565: Spatial Graph done.SG: 16 nodes and 0 edges.
2024-04-12 13:02:17.036 | INFO     | src.utils.tracklet:postprocessing_tg:324 - Post-processing done, removed 1 edges. Graph: 35 nodes, 13 edges.
2024-04-12 13:02:17.324 | INFO     | src.tracker:_test_one_epoch:261 - Iteration 565: Temporal Graph done.TG: 22 nodes and 0 edges.
2024-04-12 13:02:17.324 | INFO     | src.tracker:_test_one_epoch:264 - 【Finished inference iteration 565/568】

> IDF1   IDP   IDR   Rcll   Prcn GT MT PT ML FP FN  IDs  FM  MOTA  MOTP  IDt  IDa IDm
0       25.8% 31.5% 25.8%  99.9%  99.9% 15 15  0  0  4  4 1436   4 70.6% 0.012  584  596   1
1       30.4% 38.0% 30.4% 100.0% 100.0% 12 12  0  0  0  0 1051   0 77.8% 0.000  355  489   2
2       21.4% 27.6% 21.4% 100.0% 100.0% 15 15  0  0  0  0 2145   0 73.1% 0.003 1012  836   3
3       22.5% 28.5% 22.5% 100.0% 100.0% 13 13  0  0  0  0 1216   0 69.4% 0.016  615  550   5
OVERALL 24.6% 31.0% 24.6% 100.0% 100.0% 55 55  0  0  4  4 5848   4 72.9% 0.007 2566 2471  11

![cluster_12](https://github.com/chengche6230/ReST/assets/166817267/99c6720b-1f0d-4031-b6d2-2138f5f1ca0f)

I would kindly ask for some advice to find what is going wrong here. Thank you so much.

**Comment by chengche6230** [comment]
Hi,

Can you provide me more details, e.g. training loss, iteration, etc., about your training process? I didn't encounter such issue if the graph is well-trained.

**Comment by gucz1** [comment]
My result is the same as this problem; the reproduction clearly shows a problem.
